evm/src/cpu/kernel/tests/bignum.rs

```diff
@@ -0,0 +1,161 @@
+use anyhow::Result;
+use ethereum_types::U256;
+use rand::Rng;
+
+use crate::cpu::kernel::aggregator::KERNEL;
+use crate::cpu::kernel::interpreter::Interpreter;
+
+fn u256_to_be_limbs(x: U256) -> Vec<u8> {
+    let mut limbs = vec![0; 32];
+    x.to_big_endian(&mut limbs);
+    limbs
+}
+
+fn gen_u256_limbs<R: Rng>(rng: &mut R, num_bits: usize) -> [u64; 4] {
+    let remaining = num_bits % 64;
+    let top_limb: u64 = rng.gen_range(0..(1 << remaining));
+    if num_bits < 64 {
+        [top_limb, 0, 0, 0]
+    } else if num_bits < 128 {
+        [rng.gen(), top_limb, 0, 0]
+    } else if num_bits < 192 {
+        [rng.gen(), rng.gen(), top_limb, 0]
+    } else {
+        [rng.gen(), rng.gen(), rng.gen(), top_limb]
+    }
+}
+
+fn gen_range_u256(max: U256) -> U256 {
+    let mut rng = rand::thread_rng();
+
+    let num_bits = max.bits();
+
+    let mut x: U256 = {
+        let limbs = gen_u256_limbs(&mut rng, num_bits);
+        U256(limbs)
+    };
+    while x > max {
+        x = {
+            let limbs = gen_u256_limbs(&mut rng, num_bits);
+            U256(limbs)
+        };
+    }
+    x
+}
+
```

Comment:
These tests are a good start, but there’s a _lot_ of edge cases that are not tested.